python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/search.py

```diff
@@ -127,6 +133,41 @@ def search_facets(
         response = response["aggregateAcrossEntities"]
         return self._parse_facets(response.get("facets", []))
 
+    def list_data_product_assets(
+        self, urn: str, count: int, start: int = 0
+    ) -> SearchResponse:
+        """
+        returns a SearchResponse containing all assets in given data product (by urn)
+        """
+        variables = {
+            "urn": urn,
+            "query": "*",
+            "start": start,
+            "count": count,
+        }
+        try:
+            response = self.graph.execute_graphql(
+                self.data_product_asset_query, variables
+            )
+        except GraphError as e:
+            raise Exception("Unable to execute listDataProductAssets query") from e
+        page_results = []
+        for result in response["listDataProductAssets"]["searchResults"]:
+            entity = result["entity"]
+            entity_type = entity["type"]
+            matched_fields = {
```

Comment:
this isn't returned by this query is it? If not we can drop it

Reply:
Will double check, and drop if not returned

Reply:
is used in `_parse_dataset` so rather than refactor that i'll just pass it an empty dict